### PR TITLE
Infer NNCF backend from model type in accuracy-aware training

### PR DESCRIPTION
**Pull request by vanyalzr**

### CODE REVIEW

**Comment by vanyalzr** [comment]
Jenkins retry a build

**Comment by vshampor** [comment]
Jenkins please retry a build

File: nncf/common/accuracy_aware_training/training_loop.py [hunk 1]
```diff
@@ -24,20 +24,36 @@
...
+    try:
+        import tensorflow
+    except ImportError:
+        tensorflow = None
+
+    if torch is not None and isinstance(compression_controller.model, torch.nn.Module):
+        from nncf.torch.accuracy_aware_training.runner import PTAccuracyAwareTrainingRunner
+        return PTAccuracyAwareTrainingRunner
+
+    if tensorflow is not None and isinstance(compression_controller.model, tensorflow.Module):
```
Comment (AlexKoff88) [inline nncf/common/accuracy_aware_training/training_loop.py:49]: @alexsu52, can you please look at this part specifically and comment if it is valid?

**Review by vshampor — APPROVED** [review]

File: nncf/common/accuracy_aware_training/training_loop.py (hunk 1 above)
Comment (alexsu52) [reply nncf/common/accuracy_aware_training/training_loop.py:49]: @AlexKoff88, yes we discussed this with @vanyalzr. I can confirm that it is valid.

File: nncf/common/accuracy_aware_training/training_loop.py [hunk 2]
```diff
@@ -24,20 +24,36 @@
...
-elif __nncf_backend__ == 'TensorFlow':
-    from nncf.tensorflow.accuracy_aware_training.runner import TFAccuracyAwareTrainingRunner as \
-        AccuracyAwareTrainingRunner
 
 
 ModelType = TypeVar('ModelType')
 ADAPTIVE_COMPRESSION_CONTROLLERS = Registry('adaptive_compression_controllers')
 
 
+def get_backend_specific_training_runner_cls(compression_controller: CompressionAlgorithmController,
```
Comment (alexsu52) [inline nncf/common/accuracy_aware_training/training_loop.py:33]: Please, move this function to a separate module as you suggested.

**Review by alexsu52 — CHANGES_REQUESTED** [review]

File: nncf/common/accuracy_aware_training/training_loop.py (hunk 2 above)
Comment (vanyalzr) [reply nncf/common/accuracy_aware_training/training_loop.py]: Done

**Review by alexsu52 — APPROVED** [review]

**Comment by vanyalzr** [comment]
Jenkins please stop this build

**Comment by vanyalzr** [comment]
Jenkins please retry a build

**Comment by vanyalzr** [comment]
Jenkins please retry a build

**Comment by vanyalzr** [comment]
Jenkins please retry a build

**Comment by vanyalzr** [comment]
Jenkins please retry a build

**Comment by vanyalzr** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by vshampor** [comment]
Jenkins please stop all builds

**Comment by vshampor** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please stop all builds

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please stop all builds

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by FyodorK** [comment]
Jenkins please retry a build

**Comment by vanyalzr** [comment]
Jenkins please retry a build

**Comment by vshampor** [comment]
Jenkins please retry a build

**Comment by vanyalzr** [comment]
Jenkins please retry a build

**Comment by vshampor** [comment]
Jenkins please retry a build

**Comment by vshampor** [comment]
Jenkins please retry a build

**Comment by vanyalzr** [comment]
Jenkins please retry a build
